### DISCUSSION TITLE
Possible to activate code coverage?

### DISCUSSION THREAD

**Discussion by Tom4U** [lucono/karma-test-explorer#58] [Issues / Help] [answered]
During test runs I'd also like to have code coverage run, so that results of addons like Coverage Gutters are up-to-date. Is that possible somehow or would it be a feature request?

**Comment by lucono** [comment, marked as answer]
Karma test explorer primarily integrates with Karma as a reporter, while disabling the reporters that are setup in the project's `karma.conf` file. The rationale is that Karma Test Explorer's visual reporting of the tests is the only reporting that's relevant in the VS Code context, and disabling the other reporters in the project's karma config avoids the performance cost of running them, especially given that they are usually redundant in the VS Code context. The code coverage case here is perhaps one example where the other reporter is not redundant.

But basically, that behavior means that any code coverage tool that functions as a Karma reporter will not currently work with Karma Test Explorer.

Also, Karma Test Explorer did not always take the approach of making itself the sole reporter enabled in the modified Karma setup, but there were cases where other reporters interfered with its functioning, one example being the [BrowserStack reporter](https://github.com/karma-runner/karma-browserstack-launcher#browserstack-reporter).

[This line](https://github.com/lucono/karma-test-explorer/blob/21da962c55328a1375f34436f3ec1e9a8ba5c14b/src/frameworks/karma/config/karma-config-loader.ts#L112) is where the reporters confgured in `karma.conf` are removed, and Karma Test Explorer is added. To see if this might be relatively straight forward, try changing it to the following to see if that gets code coverage working as you would like:

```diff
    const reporters = Array.isArray(config.reporters) ? config.reporters : [];
-   reporters.splice(0, reporters.length, reporterName);
+   reporters.push(reporterName);
    config.reporters = reporters;
```

**Reply by Tom4U** [reply]
Thank for your explicit and detailed reply! Meanwhile I've found a much easier solution.

I'm using Angular and before I've had a NPM script using the `--coverage` argument. But I thought it's worth checking the options of Angular's test schematic and found out, there's a coverage option as well. 
If that is set, it works. I guess because you're using `ng` command under the hood.

Thanks again for your reply and for this great addon!!

**Reply by lucono** [reply]
That's great, I'll be adding that to the documentation.
